internal/gitinterface/log_test.go

```diff
@@ -102,4 +108,171 @@ func TestGetCommitsBetweenRange(t *testing.T) {
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		commitIDs = append(commitIDs, childRef.Hash())
+		children = append(children, childRef.Hash())
+	}
+	// creating a merge commit from the two children
+
+	commit = CreateCommitObject(testGitConfig, treeHashes[4], children, fmt.Sprintf("Test commit %v", 5), testClock)
+
+	ref, err = applyCommitAndRefreshRef(repo, commit, ref)
+	if err != nil {
+		t.Fatal(err)
+	}
+	commitIDs = append(commitIDs, ref.Hash())
+
+	// creating children for each of the two children
+
+	commit = CreateCommitObject(testGitConfig, treeHashes[3], []plumbing.Hash{children[0]}, fmt.Sprintf("Test commit %v", 4), testClock)
+
+	ref, err = applyCommitAndRefreshRef(repo, commit, ref)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	commitIDs = append(commitIDs, ref.Hash())
+
+	commit = CreateCommitObject(testGitConfig, treeHashes[5], []plumbing.Hash{children[1]}, fmt.Sprintf("Test commit %v", 6), testClock)
+
+	ref, err = applyCommitAndRefreshRef(repo, commit, ref)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	commitIDs = append(commitIDs, ref.Hash())
+
+	// Git tree with merge commit structure without its commit trees and its values :
+	//
+	//  commit 4       commit 5         commit 6
+	//    │              │  │              │
+	//    └─► commit 2 ◄─┘  └─► commit 3 ◄─┘
+	// 		      │              │
+	// 		 	  └─► commit 1 ◄─┘
+
+	allCommits, err := getCommitsFromCommitIds(commitIDs, repo)
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Run("Get all commits from merge commit 5", func(t *testing.T) {
+		commits, err := GetCommitsBetweenRange(repo, commitIDs[3], plumbing.ZeroHash)
+		assert.Nil(t, err)
+		expectedCommits := []*object.Commit{allCommits[3], allCommits[2], allCommits[1], allCommits[0]}
+		sort.Slice(expectedCommits, func(i, j int) bool {
+			return expectedCommits[i].ID().String() < expectedCommits[j].ID().String()
+		})
+
+		assert.Equal(t, expectedCommits, commits)
+		assert.Nil(t, err)
+	})
+	t.Run("Get all commits from merge commit 6", func(t *testing.T) {
+		commits, err := GetCommitsBetweenRange(repo, commitIDs[3], plumbing.ZeroHash)
+		assert.Nil(t, err)
+		expectedCommits := []*object.Commit{allCommits[3], allCommits[2], allCommits[1], allCommits[0]}
+		sort.Slice(expectedCommits, func(i, j int) bool {
+			return expectedCommits[i].ID().String() < expectedCommits[j].ID().String()
+		})
+		assert.Equal(t, expectedCommits, commits)
+		assert.Nil(t, err)
+	})
+}
+
+func getCommitsFromCommitIds(commitIDs []plumbing.Hash, repo *git.Repository) ([]*object.Commit, error) {
+	allCommits := make([]*object.Commit, 0, len(commitIDs))
+	for _, commitID := range commitIDs {
+		commit, err := GetCommit(repo, commitID)
+		if err != nil {
+			return nil, err
+		}
+
+		allCommits = append(allCommits, commit)
+	}
+	return allCommits, nil
+}
+
+func applyCommitAndRefreshRef(repo *git.Repository, commit *object.Commit, ref *plumbing.Reference) (*plumbing.Reference, error) {
```

Comment:
1. Do we need to return a new plumbing.Reference? Since we're passing in the pointer of the ref, just updating it should do?
2. If this is only used in tests, let's pass in `t`, register this as a helper `t.Helper()`, and abort in case of errors `t.Fatal`. If 1 holds and with this change, we don't need to return anything at all here.

Reply:
Also, it'd be neat to track other places this helper could be used. I think there are certainly other tests that could reuse this.

Reply:
They don't need to be updated in this PR though.

Reply:
Fixed